engine/engine/content/builtins/scripts/mobdebug.lua

```diff
@@ -589,6 +641,7 @@ local function debug_hook(event, line)
     stack_level = stack_level + 1
   elseif event == "return" or event == "tail return" then
     stack_level = stack_level - 1
+    if mobdebug.on_return_hook then mobdebug.on_return_hook() end
```

Comment:
it's important to restore lightweight hook when we exit stack
we can't do it earlier because user should be able to set breakpoints even if current paused breakpoint in the middle of a function (as you remember lightweight breakpoint takes into account only function calls)